pkg/intercp/catalog/catalog.go

```diff
@@ -41,6 +42,19 @@ func Leader(ctx context.Context, catalog Catalog) (Instance, error) {
 	return Instance{}, ErrNoLeader
 }
 
+func InstanceOfID(ctx context.Context, catalog Catalog, id string) (Instance, error) {
```

Comment:
`InstanceID` would suggest that it _is_ an Instance ID or we try to retrieve `InstanceID`. Here we are retrieving Instance by ID? Would `InstanceByID` be a better name?

I used `Of` because we use `SomethingByKey` usually when we have map of Key -> Something